src/cmds/qstat.c

```diff
@@ -598,55 +599,81 @@ trunc_value(char *value, int len, int wide_len, int wide)
 static void
 prt_nodes(char *nodes, int no_newl)
 {
-	int  i;
+	int  i, len;
 	char linebuf[78];
-	char *stp;
-	int   vnodeid = 0;
+	char *rest = NULL;
+	char *saveptr = NULL;
+ 	char *token = NULL;
+	char *token_cp = NULL;
+	char *subtoken = NULL;
+	char *node_name = NULL;
+	char *node_name_bkp = NULL;
+	char *chunk = NULL;
+	struct sockaddr_in check_ip;
+	int ret = 0;
 
 	if ((nodes == NULL) || (*nodes == '\0'))
 		return;
 
 	i = 0;
-	stp = nodes;
-	while (*nodes != '\0') {
-		if (*stp == '[')
```

Comment:
It looks like you removed the check for brackets, which is what "i" was being used for. That's the source of my confusion. I think the result is a slight change in behavior, and I don't think that's what we want.

Reply:
The check here for '[' and ']' made sure that we did not remove the brackets from the the <host> of the exec_host string (<host>/<t1>*<p1> + <host1>/<t2>*<p2>). After the fix too, we are retaining the brackets. We are not specifically checking for brackets because if <host> is not a valid IP address, we are using the value of <host> as is.
I have attached the log which shows the behavior of qstat -n before and after the fix. 
[qstat_with_sq_bracket.txt](https://github.com/PBSPro/pbspro/files/3931568/qstat_with_sq_bracket.txt)